src/file_pipe_log/log_file.rs

```diff
@@ -148,59 +141,45 @@ impl<F: FileSystem> LogFileWriter<F> {
 }
 
 /// Build a file reader.
-///
-/// Attention please, the reader do not need a specified `[LogFileFormat]` from
-/// users.
-///
-/// * `handle`: standard handle of a log file.
-/// * `format`: if `[None]`, reloads the log file header and parse
-/// the relevant `LogFileFormat` before building the `reader`.
 pub(super) fn build_file_reader<F: FileSystem>(
     system: &F,
     handle: Arc<F::Handle>,
-    format: Option<LogFileFormat>,
 ) -> Result<LogFileReader<F>> {
     let reader = system.new_reader(handle.clone())?;
-    LogFileReader::open(handle, reader, format)
+    Ok(LogFileReader::open(handle, reader))
 }
 
 /// Random-access reader for log file.
 pub struct LogFileReader<F: FileSystem> {
-    format: LogFileFormat,
     handle: Arc<F::Handle>,
     reader: F::Reader,
 
     offset: u64,
 }
 
 impl<F: FileSystem> LogFileReader<F> {
-    fn open(
-        handle: Arc<F::Handle>,
-        reader: F::Reader,
-        format: Option<LogFileFormat>,
-    ) -> Result<Self> {
-        match format {
-            Some(fmt) => Ok(Self {
-                format: fmt,
-                handle,
-                reader,
-                // Set to an invalid offset to force a reseek at first read.
-                offset: u64::MAX,
-            }),
-            None => {
-                let mut reader = Self {
-                    format: LogFileFormat::default(),
-                    handle,
-                    reader,
-                    // Set to an invalid offset to force a reseek at first read.
-                    offset: u64::MAX,
-                };
-                reader.parse_format()?;
-                Ok(reader)
-            }
+    fn open(handle: Arc<F::Handle>, reader: F::Reader) -> LogFileReader<F> {
+        Self {
+            handle,
+            reader,
+            // Set to an invalid offset to force a reseek at first read.
+            offset: u64::MAX,
```

Comment:
I've got the purpose that u refactor the `LogFileReader` and its related method `build_file_reader`. But referring to the design on `LogFileWriter`, I still recommend to make the `parse_format` integrated into the processing of `LogFileReader::open()`, rather than `open()` -> manually `parse_format` by callers.

Reply:
My change removes the `format` stored inside reader, because reader doesn't need a format to function. The writer, on the other hand, needs to do format-specific initialization, so we can't refactor it like reader.